src/rotest/api/resource_control/lock_resources.py

```diff
@@ -0,0 +1,117 @@
+# pylint: disable=unused-argument, no-self-use, too-many-locals
+import httplib
+from datetime import datetime
+
+from django.db import transaction
+from django.db.models.query_utils import Q
+from django.core.exceptions import FieldError
+from django.contrib.auth import models as auth_models
+from swaggapi.api.builder.server.response import Response
+from swaggapi.api.builder.server.exceptions import BadRequest
+from swaggapi.api.builder.server.request import DjangoRequestView
+
+from rotest.management.common.utils import get_username
+from rotest.management.common.json_parser import JSONParser
+from rotest.api.common.models import LockResourcesParamsModel
+from rotest.management.common.resource_descriptor import ResourceDescriptor
+from rotest.api.common.responses import (InfluencedResourcesResponseModel,
+                                         BadRequestResponseModel)
+
+
+class LockResources(DjangoRequestView):
+    """Lock the given resources one by one.
+
+    Note:
+        If one of the resources fails to lock, all the resources
+        that has been locked until that resource will be released.
+    """
+    URI = "resources/lock_resources"
+    DEFAULT_MODEL = LockResourcesParamsModel
+    DEFAULT_RESPONSES = {
+        httplib.OK: InfluencedResourcesResponseModel,
+        httplib.BAD_REQUEST: BadRequestResponseModel
+    }
+    TAGS = {
+        "post": ["Resources"]
+    }
+
+    def _lock_resource(self, resource, user_name):
+        """Mark the resource as locked by the given user.
+
+        For complex resource, marks also its sub-resources as locked by the
+        given user.
+
+        Note:
+            The given resource *must* be available.
```

Comment:
What does it mean that it must? If not, it's the programmer's mistake?

Reply:
This is copy pasted from the original code.
I dont know. maybe @UnDarkle knows?
maybe it messes up the user cleanup.. i am not sure

Reply:
Fix it